src/main/daml/Daml/Finance/Settlement/Instruction.daml

```diff
@@ -279,9 +279,21 @@ discloseAccountHelper f (context, settlers) account isInstructor = do
   f (context, settlers) viewer disclosers account
 
 -- | HIDE
-settlementIsDue : Optional Time -> Update ()
-settlementIsDue settlementTime =
-  whenSome settlementTime $ \t -> do
-    now <- getTime
-    assertMsg ("settlement time " <> show t <> " must be due (now is " <> show now <> ")") $
-      now >= t
+settlementIsDue : Optional SettlementTime -> Parties -> Optional (ContractId TimeObservable.I) ->
+  Update ()
+settlementIsDue oSettlementTime actors oTimeObservableCid =
+  whenSome oSettlementTime $ \SettlementTime{due; providers; id} -> do
+    case oTimeObservableCid of
+      Some timeObservableCid -> do
+        timeObservable <- exercise timeObservableCid TimeObservable.GetView with
+          viewer = L.head $ S.toList actors
```

Comment:
Is this guaranteed to work for any `actor` in `actors` ?

Reply:
yes, anyone (having visibility) can call the `GetView` choice